src/VisualStudio/Core/Next/Remote/JsonRpcSession.cs

```diff
@@ -65,9 +65,23 @@ internal class JsonRpcSession : RemoteHostClient.Session
 
         private async Task InitializeAsync()
         {
-            // all roslyn remote service must based on ServiceHubServiceBase which implements Initialize method
-            await _snapshotClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
-            await _serviceClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
+            CancellationToken.ThrowIfCancellationRequested();
+
+            try
+            {
+                // all roslyn remote service must based on ServiceHubServiceBase which implements Initialize method
+                await _snapshotClient.InvokeAsync(WellKnownServiceHubServices.ServiceHubServiceBase_Initialize, _currentSessionId, PinnedScope.SolutionChecksum.ToArray()).ConfigureAwait(false);
```

Comment:
so, except the session, who else is using that JsonRpcClient? Session is the one people see.